client.go

```diff
@@ -64,87 +56,131 @@ type Client struct {
 	AWSPrivatelink                  *AWSPrivatelinkHandler
 }
 
-// GetUserAgentOrDefault configures a default userAgent value, if one has not been provided.
-func GetUserAgentOrDefault(userAgent string) string {
-	if userAgent != "" {
-		return userAgent
+type Option func(*clientParameters)
+
+func WithHTTPClient(c *http.Client) Option {
+	return func(cp *clientParameters) {
+		cp.httpClient = c
 	}
-	return "aiven-go-client/" + Version()
 }
 
-// NewMFAUserClient creates a new client based on email, one-time password and password.
-func NewMFAUserClient(email, otp, password string, userAgent string) (*Client, error) {
-	c := &Client{
-		Client:    buildHttpClient(),
-		UserAgent: GetUserAgentOrDefault(userAgent),
+func WithAPIUrl(url string) Option {
+	return func(cp *clientParameters) {
+		cp.apiUrl = url
 	}
+}
 
-	bts, err := c.doPostRequest("/userauth", authRequest{email, otp, password})
-	if err != nil {
-		return nil, err
+func WithTokenAuth(token string) Option {
+	return func(cp *clientParameters) {
+		cp.authMethod = tokenAuth{token}
 	}
+}
 
-	var r authResponse
-	if err := checkAPIResponse(bts, &r); err != nil {
-		return nil, err
+func WithMFAAuth(email, password, otp string) Option {
+	return func(cp *clientParameters) {
+		cp.authMethod = mfaAuth{email: email, otp: otp, password: password}
 	}
+}
 
-	return NewTokenClient(r.Token, userAgent)
+func WithUserAuth(email, password string) Option {
+	return func(cp *clientParameters) {
+		cp.authMethod = mfaAuth{email: email, password: password}
+	}
 }
 
-// NewUserClient creates a new client based on email and password.
-func NewUserClient(email, password string, userAgent string) (*Client, error) {
-	return NewMFAUserClient(email, "", password, userAgent)
+func WithUserAgent(userAgent string) Option {
+	return func(cp *clientParameters) {
+		cp.userAgent = userAgent
+	}
 }
 
-// NewTokenClient creates a new client based on a given token.
-func NewTokenClient(key string, userAgent string) (*Client, error) {
-	c := &Client{
-		APIKey:    key,
-		Client:    buildHttpClient(),
-		UserAgent: GetUserAgentOrDefault(userAgent),
+func WithRetries(retryCount uint, retryBackoff time.Duration) Option {
```

Comment:
It looks that we never use this function, do we need it?

Reply:
Its intended for clients to configure retries/retry timeouts